.semaphore/semaphore.yml

```diff
@@ -0,0 +1,26 @@
+version: v1.0
+name: AppSignal sql_lexer
+agent:
+  machine:
+    type: e1-standard-2
+  containers:
+    - name: main
+      image: registry.semaphoreci.com/rust:1.51
+blocks:
+  - name: 'Default'
+    task:
+      jobs:
+        - name: Formatter and linter
+          commands:
+            - checkout
+            - rustup component add rustfmt clippy
+            - cache restore clippy-${SEMAPHORE_GIT_BRANCH}-${LOCK_CHECKSUM}
+            - cargo fmt --verbose --all -- --check
+            - cargo clippy
+            - cache store clippy-${SEMAPHORE_GIT_BRANCH}-${LOCK_CHECKSUM}
```

Comment:
This prompts an error. I don't know why it's not failing the build. You need to  specify the path you want to cache.

```
Usage:
      cache store (lookup cachable elements in the project and cache them e.g vendor/bundle)
      cache store [key] [local-path]
Example:
      cache store bundle-$SEMAPHORE_GIT_BRANCH-$(checksum Gemfile.lock) vendor/bundle
! [13:08 16/09/2021]: Incorrect number of arguments!
```

Reply:
I just learned that Rust libraries, as this one, should gitignore the `Cargo.lock` file, so we don't have anything to cache here.